independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/JarClassPathElement.java

```diff
@@ -72,9 +73,10 @@ public String getPath() {
                         @Override
                         public URL getUrl() {
                             try {
-                                return new URL("jar", null, jarPath.getProtocol() + ":" + jarPath.getPath() + "!/" + name);
+                                return new URL("jar", null,
+                                        jarPath.getProtocol() + ":" + jarPath.getPath() + "!/" + JarEntries.getRealName(res));
```

Comment:
This change makes the `io.quarkus.webjar.locator.test.WebJarLocatorDevModeTest` test fail in JDK 8. I'm investigating why that happens